src/Jellyfin.Maui/Services/NavigationService.cs

```diff
@@ -103,29 +102,29 @@ public void NavigateToItemView(BaseItemDto item)
         switch (item.Type)
         {
             case BaseItemKind.CollectionFolder:
-                Navigate<LibraryPage, LibraryViewModel>(item);
+                Navigate<LibraryViewModel>(item);
                 break;
             default:
-                Navigate<ItemPage, ItemViewModel>(item);
+                Navigate<ItemViewModel>(item);
                 break;
         }
     }
 
-    private void Navigate<TPage, TViewModel>(BaseItemDto item)
+    private void Navigate<TViewModel>(BaseItemDto item)
         where TViewModel : BaseItemViewModel
-        where TPage : BaseContentIdPage<TViewModel>
     {
-        if (_navigationPage is null)
+        var pageType = typeof(NavigationService).Assembly.GetTypes()
+            .FirstOrDefault(x => typeof(BaseContentPage<TViewModel>).IsAssignableFrom(x));
+
+        if (pageType == null)
         {
-            NavigateHome();
+            Debug.WriteLine($"The ViewModel '{typeof(TViewModel).Name}' is not associated with a BaseContentPage<>");
             return;
         }
 
         Application.Current?.Dispatcher.Dispatch(() =>
         {
-            var resolvedView = InternalServiceProvider.GetService<TPage>();
-            resolvedView.Initialize(item);
-            _navigationPage.PushAsync(resolvedView, true).SafeFireAndForget();
+            Shell.Current.GoToAsync($"{pageType.Name}?args={Uri.EscapeDataString(JsonSerializer.Serialize(item))}", true);
```

Comment:
1. This is just a transitory step, I did not want to change the `BaseItemViewModel ` class in order to have few changes to review.
2. Object-based navigation (either with IDictionary<string, object> or via Json serialization) has its limits: if its too big the URL will trimmed. It 'works' now because the passed `BaseItemDto` is not the full representation of the object (the 'full' object is obtained via `_libraryService.GetItemAsync(Item.Id))`.
3. So next step is to have `BaseItemViewModel.Initialize(Guid)` instead of `BaseItemViewModel.Initialize(BaseItemDto)`  .

Reply:
That makes sense, and is how it was originally implemented